backend/dataall/cdkproxy/stacks/policies/sagemaker.py

```diff
@@ -15,126 +36,170 @@ def get_statements(self):
+                resources=[f'arn:aws:sagemaker:{self.region}:{self.account}:app/*/*']
+            ),
+            iam.PolicyStatement(
+                # sid="SageMakerCreatePresignedDomainUrl",
+                effect=iam.Effect.ALLOW,
+                actions=['sagemaker:CreatePresignedDomainUrl'],
+                resources=[f'arn:aws:sagemaker:{self.region}:{self.account}:user-profile/*/*'],
+                conditions={
+                    'StringEquals': {
+                        f'sagemaker:ResourceTag/{self.tag_key}': [self.tag_value]
+                    },
+                },
+            ),
+            iam.PolicyStatement(
+                # sid="SageMakerManageGenericResources",
+                effect=iam.Effect.ALLOW,
+                actions=[
+                    'sagemaker:Delete*',
+                    'sagemaker:Update*',
+                    'sagemaker:Start*',
+                    'sagemaker:Stop*',
+                    'sagemaker:InvokeEndpoint',
+                    'sagemaker:InvokeEndpointAsync'
+                ],
                 resources=[
-                    f'arn:aws:sagemaker:{self.region}:{self.account}:endpoint/*'
+                    f'arn:aws:sagemaker:{self.region}:{self.account}:*/{self.resource_prefix}*',
+                    f'arn:aws:sagemaker:{self.region}:{self.account}:*/{self.resource_prefix}*/*',
                 ],
                 conditions={
                     'StringEquals': {
-                        f'aws:ResourceTag/{self.tag_key}': [self.tag_value]
-                    }
+                        f'aws:ResourceTag/{self.tag_key}': [self.tag_value],
+                    },
                 },
             ),
+            # Logging and support permissions
             iam.PolicyStatement(
+                #sid="SageMakerLogging",
+                effect=iam.Effect.ALLOW,
                 actions=[
                     'logs:CreateLogGroup',
                     'logs:CreateLogStream',
-                    'logs:PutLogEvents'],
+                    'logs:PutLogEvents'
+                ],
                 resources=[
                     f'arn:aws:logs:{self.region}:{self.account}:log-group:/aws/sagemaker/*',
                     f'arn:aws:logs:{self.region}:{self.account}:log-group:/aws/sagemaker/*:log-stream:*',
                 ]
             ),
             iam.PolicyStatement(
+                #sid="SageMakerSupport",
+                effect=iam.Effect.ALLOW,
                 actions=[
                     'ecr:GetAuthorizationToken',
                     'ecr:BatchCheckLayerAvailability',
                     'ecr:GetDownloadUrlForLayer',
-                    'ecr:BatchGetImage'],
-                resources=[
-                    '*'
-                ]
+                    'ecr:BatchGetImage',
+                    'servicecatalog:ListAcceptedPortfolioShares',
+                    'servicecatalog:ListPrincipalsForPortfolio',
+                ],
+                resources=['*']
```

Comment:
Not sure if we should set wildcard here. Especially, I am concerned about `GetAuthorizationToken`